src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -188,6 +187,52 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
                 var nullMetricsLogger = s.GetService<ILogger<NullMetricsPublisher>>();
                 return new NullMetricsPublisher(nullMetricsLogger);
             });
+
+            services.AddSingleton<IMeshInitServiceClient>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                if (environment.IsLinuxConsumption())
+                {
+                    var httpClient = s.GetService<HttpClient>();
+                    var logger = s.GetService<ILogger<MeshInitServiceClient>>();
+                    return new MeshInitServiceClient(httpClient, environment, logger);
+                }
+
+                var nullLogger = s.GetService<ILogger<NullMeshInitServiceClient>>();
+                return new NullMeshInitServiceClient(nullLogger);
+            });
+
+            services.AddSingleton<LinuxFunctionExecutionActivityPublisher>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                var logger = s.GetService<ILogger<LinuxFunctionExecutionActivityPublisher>>();
```

Comment:
Why isn't this service conditional like the others? Won't this lead to failures in non-Linux, since you're calling GetService<IMeshInitServiceClient> which will be null?

Reply:
for non-linux scenarions null is what we want. returning null explicitly in such cases.